qiskit_experiments/database_service/utils.py

```diff
@@ -74,6 +75,47 @@ def utc_to_local(utc_dt: datetime) -> datetime:
     return local_dt
 
 
+def objs_to_zip(filenames: List[str], contents: List[str]) -> bytes:
+    """Write a list of files and contents to a zipped file.
+
+    Args:
+        filenames: List of names for the files to be zipped.
+        contents: List of file contents to be zipped in string form.
+
+    Returns:
+        A bytes object containing the zipped files.
+    """
+    zip_buffer = io.BytesIO()
+
+    with zipfile.ZipFile(zip_buffer, "w", zipfile.ZIP_DEFLATED, False) as zip_file:
+        for file_name, data in zip(filenames, contents):
+            zip_file.writestr(file_name, data)
+
+    zip_buffer.seek(0)
+    return zip_buffer
+
+
+def zip_to_objs(zip_bytes: bytes) -> List[str]:
+    """Extract files from a zipped buffer as strings.
+
+    Args:
+        zip_bytes: Bytes object representing the zipped file.
+
+    Returns:
+        A list of strings extracted from the zip file buffer.
+    """
+    output_strings = []
+    zip_buffer = io.BytesIO(zip_bytes)
+
+    with zipfile.ZipFile(zip_buffer, "r") as zip_file:
+        for file_name in zip_file.namelist():
+            with zip_file.open(file_name) as file:
+                data = file.read().decode("utf-8")
+                output_strings.append(data)
+
+    return output_strings
```

Comment:
I think it's better to yield here for memory footprint because each object can be heavy in artifact.